pkg/operator/ceph/csi/spec.go

```diff
@@ -1092,3 +1093,7 @@ func getPrefixFromArg(arg string) (string, bool) {
 	}
 	return "", false
 }
+
+func EnableCSIOperator() bool {
+	return enableCSIOperator && !IsHolderEnabled()
```

Comment:
lets not use local variable to check whether csi driver is enabled or not, because this is a shared function between multiple controllers, use config map get to ensure that we are always consistent with result

Reply:
@Madhu-1 at https://github.com/rook/rook/pull/14489/commits/dea6123a29b53843dfff4f700c5c12221bafeda6#diff-6a89cde4909fd851e48da3acc4dd4628c2fdbd68c40226d0b6e789f8b36a62eeR186-R189 we are reading from configmap only. And whenever the value is updated in configmap that part of the code is always triggered so do we need to use from configmap here?

Reply:
Yes thats correct, this works if the csi controller is initialized and already read the configmap and updated the local global variable. This works for most of the cases but for very corner cases, we don't get the expected result.

Reply:
In all cases today IIRC, csi controller is reconciled, so do we really need to get the configmap and read the field?
I'm just trying to understand before dong the changes

Reply:
This is the current bug we have today and chances of hitting is less but we can leave with it just letting you know

Reply:
okay understood, thanks for letting me know.